component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java

```diff
@@ -592,80 +590,7 @@ protected void info(final String msg) {
-        final String m2Repo = System.getProperty("maven.repository");
-        if (!"global".equals(m2Repo)) {
-            final Path localM2 =
-                    PathFactory.get(System.getProperty("osgi.configuration.area", "")).resolve(".m2/repository");
-            if (java.nio.file.Files.exists(localM2)) {
-                return localM2;
-            }
-        }
-        log.debug("[findStudioM2] could not get studio config");
-        return findDefaultM2();
-    }
-
-    private static Path findDefaultM2() {
-        // check out settings.xml first
-        final Path settings = PathFactory
-                .get(System
-                        .getProperty("talend.component.manager.m2.settings",
-                                System.getProperty("user.home") + "/.m2/settings.xml"));
-        log.debug("[findDefaultM2] searching for localRepository location in {}", settings);
-        if (java.nio.file.Files.exists(settings)) {
-            try {
-                String localM2RepositoryFromSettings = null;
-                // do not introduce a xml parser so will do with what we have...
-                final String content = new String(java.nio.file.Files.readAllBytes(settings), StandardCharsets.UTF_8);
-                final Pattern comments = Pattern.compile("(<!--.*?-->)", Pattern.DOTALL);
-                final Pattern emptyLines = Pattern.compile("^\\s*$|\\n|\\r\\n");
-                final Pattern localRepo =
-                        Pattern.compile(".*<localRepository>(.+)</localRepository>.*", Pattern.CASE_INSENSITIVE);
-                // some cleanups
-                String stripped = comments.matcher(content).replaceAll("");
-                stripped = emptyLines.matcher(stripped).replaceAll("");
-                // localRepository present?
-                final Matcher m = localRepo.matcher(stripped);
-                if (!m.matches()) {
-                    log.debug("[findDefaultM2] localRepository not defined in settings.xml");
-                } else {
-                    localM2RepositoryFromSettings = m.group(1).trim();
-                    if (localM2RepositoryFromSettings != null && !localM2RepositoryFromSettings.isEmpty()) {
-                        final Path settingsM2 = PathFactory.get(localM2RepositoryFromSettings);
-                        if (java.nio.file.Files.exists(settingsM2)) {
-                            return settingsM2;
-                        }
-                        log.warn("[findDefaultM2] localRepository points to a non existent path: {}", settingsM2);
-                    }
-                }
-            } catch (final Exception ignore) {
-                // fallback on default local path
-            }
-        }
-        log.debug("[findDefaultM2] fallback to user's default repository");
-        return PathFactory.get(System.getProperty("user.home", "")).resolve(".m2/repository");
+        return new MavenRepositoryDefaultResolver().discover();
```

Comment:
Very nice to put code out of this huge class.
Even better with a kind indirection (ServiceLoader for example), to avoid a hard dependency.
(But great necessary first step)

Reply:
will implement SPI on next release due to remaining time to upcoming release.